File: src/main/java/net/alexsannd/HUD.java
package net.alexsannd;

import java.awt.*;

public class HUD {
    private int score;
    private int highScore;
    private int level;
    private int width;
    private int height;
    private int x;
    private int y;

    public HUD(int x, int y, int width, int height) {
        this.width = width;
        this.height = height;
        this.x = x;
        this.y = y;
        this.score = 0;
        this.highScore = 0;
        this.level = 1;
    }

    public void paint(Graphics g) {
        g.setColor(new Color(50, 50, 50));
        g.fillRect(x, y, width, height);
        g.setColor(Color.WHITE);
        g.setFont(new Font("Arial", Font.BOLD, 20));
        g.drawString("Score: " + score, x + 10, y + 20);
        g.drawString("High Score: " + highScore, x + 10, y + 40);
        g.drawString("Level: " + level, x + 10, y + 60);
    }

    public void setScore(int score) {
        this.score = score;
        if (this.score > highScore) {
            highScore = this.score;
        }
    }
    public void addScore(int score) {
        this.score += score;
        if (this.score > highScore) {
            highScore = this.score;
        }

    }

    public void setHighScore(int highScore) {
        this.highScore = highScore;
    }

    public void setLevel(int level) {
        this.level = level;
    }
}

File: src/main/java/net/alexsannd/HUD.kt
package net.alexsannd

import java.awt.Color
import java.awt.Font
import java.awt.Graphics

class HUD(private val x: Int, private val y: Int, private val width: Int, private val height: Int) {
    private var score = 0
    private var highScore = 0
    private var level = 1

    fun paint(g: Graphics) {
        g.color = Color(50, 50, 50)
        g.fillRect(x, y, width, height)
        g.color = Color.WHITE
        g.font = Font("Arial", Font.BOLD, 20)
        g.drawString("Score: $score", x + 10, y + 20)
        g.drawString("High Score: $highScore", x + 10, y + 40)
        g.drawString("Level: $level", x + 10, y + 60)
    }

    fun setScore(score: Int) {
        this.score = score
        if (this.score > highScore) {
            setHighScore(this.score)
        }
    }

    fun addScore(score: Int) {
        this.score += score
        if (this.score > highScore) {
            setHighScore(this.score)
        }
    }

    fun setHighScore(highScore: Int) {
        this.highScore = highScore
    }

    fun setLevel(level: Int) {
        this.level = level
    }
}
